python/tvm/topi/arm_cpu/qnn.py

```diff
@@ -368,3 +389,136 @@ def kernel_ptr(buffer, c, offset=0):
 def schedule_qnn_depthwise_conv2d(_attrs, _outs, _target):
     """Schedule function for qnn.depthwise_conv2d."""
     return None
+
+
+def _make_unrolled_conv2d_primfunc(
+    output_dimensions: Tuple[int, int, int],
+    buffer_shapes: Tuple[Tuple, Tuple, Tuple, Tuple, Tuple],
+    function_names: Dict[Tuple, str],
+    function_code: str,
+    ptr_gens: Tuple[Callable, Callable],
+    output_layout="NHWC",
+):
+    out_height, out_width, out_channels = output_dimensions
+    data_shape, kernel_shape, bias_shape, scale_shape, output_shape = buffer_shapes
+    data_ptr, kernel_ptr = ptr_gens
+
+    def output_ptr(output, y, c):
```

Comment:
nit: inconsistent use of leading '_' for internal functions

Reply:
My understanding was that `_` should be used for global internal functions, but that `_` should not be used for local functions. Correct me if I'm wrong, though.

Reply:
I just noticed that inside the `def _make_unrolled_conv2d_primfunc` block there is the following:
```
    def output_ptr
    def _make_row_call
    def biased_quantized_conv2d
```

But I may be missing something and I am by no means an expert on the subject 😆

Reply:
Oops, you're right - `_make_row_call` should be `make_row_call`. Thanks!